src/libcore/alloc.rs

```diff
@@ -21,10 +21,47 @@ use mem;
 use usize;
 use ptr::{self, NonNull};
 
+extern {
+    /// An opaque, unsized type. Used for pointers to allocated memory.
+    ///
+    /// This type can only be used behind a pointer like `*mut Void` or `ptr::NonNull<Void>`.
+    /// Such pointers are similar to C’s `void*` type.
+    pub type Void;
+}
+
+impl Void {
+    /// Similar to `std::ptr::null`, which requires `T: Sized`.
+    pub fn null() -> *const Self {
+        0 as _
+    }
+
+    /// Similar to `std::ptr::null_mut`, which requires `T: Sized`.
+    pub fn null_mut() -> *mut Self {
+        0 as _
+    }
+}
+
+/// Convert from a return value of GlobalAlloc::alloc to that of Alloc::alloc
+impl From<*mut Void> for Result<NonNull<Void>, AllocErr> {
```

Comment:
Since this and the impl below are insta-stable, could they be inherent methods for now or free functions?

Reply:
Aren’t these conversions effectively unstable if both types are unstable?

Reply:
I wanted functions on `Void` but hit https://github.com/rust-lang/rust/issues/49659.

Reply:
Ah ok that's true, if it's all functionally unstable anyway then it seems fine to leave as-is